compiler/ilgen/IlBuilder.cpp

```diff
@@ -516,16 +516,17 @@ IlBuilder::appendBlock(TR::Block *newBlock, bool addEdge)
    _currentBlock = newBlock;
    }
 
-
 void
-IlBuilder::AppendBuilder(TR::IlBuilder *builder)
+IlBuilder::AppendBuilder(TR::IlBuilder *builder, TR::Block *blockThrowsException)
    {
-   ILB_REPLAY("%s->AppendBuilder(%s);", REPLAY_BUILDER(this), REPLAY_BUILDER(builder));
+   ILB_REPLAY("%s->AppendBuilder(%s) isExceptionBuilder: %d;", REPLAY_BUILDER(this), REPLAY_BUILDER(builder), isExceptionBuilder);
    TR_ASSERT(builder->_partOfSequence == false, "builder cannot be in two places");
 
    builder->_partOfSequence = true;
    _sequenceAppender->add(builderEntry(builder));
-   if (_currentBlock != NULL)
+   if (blockThrowsException)
```

Comment:
I am not sure your logic is right here, but possibly I haven't caught the use case yet. If you add the exception edge, where is control supposed to flow from the _currentBlock when you enter AppendBuilder to the "appendNoFallThroughBlock" that's added at the end? You add an edge from the builder to the new _currentBlock, but the old _currentBlock just seems to fall off without any control flow edge?